File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03241/p03241_num8_parsed.cpp

#include <iostream>
#include <vector>

using namespace std;

int gcd(int a, int b) {
    if (b == 0)
        return a;
    return gcd(b, a % b);
}

int main() {
    int N, M;
    cin >> N >> M;

    // Find the maximum possible value of the greatest common divisor 
    // by checking the factors of M and selecting the largest factor that is less than or equal to M/N.
    int max_gcd = 1;
    for (int i = 1; i * i <= M; ++i) {
        if (M % i == 0) {
            int factor1 = i;
            int factor2 = M / i;
            if (factor1 <= M / N)
                max_gcd = max(max_gcd, factor1);
            if (factor2 <= M / N)
                max_gcd = max(max_gcd, factor2);
        }
    }

    cout << max_gcd << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03241/p03241_num8_parsed.py
import math

def gcd_of_sequence(N, M):
    # Find the largest possible value for the greatest common divisor
    max_gcd = min(N, M)
    
    # Find the largest divisor of M that is less than or equal to N
    for i in range(max_gcd, 0, -1):
        if M % i == 0:
            return i
    
    return 1

# Get the input values of N and M
N, M = map(int, input().split())

# Find the maximum possible value of the greatest common divisor
max_gcd = gcd_of_sequence(N, M)

# Print the result
print(max_gcd)